packages/flutter_tools/lib/src/commands/build_ipa.dart

```diff
@@ -0,0 +1,154 @@
+// found in the LICENSE file.
+
+import 'package:file/file.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../application_package.dart';
+import '../base/common.dart';
+import '../base/logger.dart';
+import '../base/platform.dart';
+import '../base/process.dart';
+import '../build_info.dart';
+import '../convert.dart';
+import '../globals.dart' as globals;
+import '../ios/xcodeproj.dart';
+import '../runner/flutter_command.dart' show FlutterCommandResult;
+import 'build.dart';
+
+/// Builds an iOS .ipa package for App Store submission.
+///
+/// A wrapper around "xcodebuild -exportArchive". Can only be run on a macOS host.
+class BuildIPACommand extends BuildSubCommand {
+  BuildIPACommand({
+    @visibleForTesting Platform platform,
+    @visibleForTesting FileSystem fileSystem,
+    @visibleForTesting Logger logger,
+    @visibleForTesting ProcessManager processManager,
+    @visibleForTesting XcodeProjectInterpreter xcodeProjectInterpreter,
+  })  : _platform = platform,
+        _xcodeProjectInterpreter = xcodeProjectInterpreter,
+        _fileSystem = fileSystem,
+        _logger = logger,
+        _processUtils =
+            ProcessUtils(processManager: processManager, logger: logger) {
+    argParser.addOption(
+      'export-options-plist',
+      valueHelp: 'ExportOptions.plist',
+      help:
+          'Path to the "-exportOptionsPlist" file passed to Xcode. See "man xcodebuild" for available keys.',
+    );
+    // Flavors may have different target names, and therefore different archive output paths.
+    usesFlavorOption();
+  }
+
+  Platform _platform;
+  XcodeProjectInterpreter _xcodeProjectInterpreter;
+  FileSystem _fileSystem;
+  Logger _logger;
+  ProcessUtils _processUtils;
+
+  @override
+  final String name = 'ipa';
+
+  @override
+  final String description =
+      'Build an iOS App Store package after running "flutter build xcarchive" (Mac OS X host only).';
+
+  String get exportOptionsPlist => stringArg('export-options-plist');
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    // Populate the "globals" that are not injected from tests.
+    _platform ??= globals.platform;
+    _xcodeProjectInterpreter ??= globals.xcodeProjectInterpreter;
+    _fileSystem ??= globals.fs;
+    _logger ??= globals.logger;
+    _processUtils ??=
+        ProcessUtils(processManager: globals.processManager, logger: _logger);
+
+    if (!_platform.isMacOS) {
+      throwToolExit('Building for iOS is only supported on macOS.');
+    }
+
+    if (exportOptionsPlist == null) {
+      throwToolExit(
+          '--export-options-plist file is required. See "man xcodebuild" for available keys.');
+    }
+
+    final FileSystemEntityType type = _fileSystem.typeSync(exportOptionsPlist);
+    if (type == FileSystemEntityType.notFound) {
+      throwToolExit(
+          '"$exportOptionsPlist" property list does not exist. See "man xcodebuild" for available keys.');
```

Comment:
I don't actually see the available keys in the manual (which is really long). We might need to do a bit more hand-holding for pure-flutter users.

Reply:
You're right, it's in the `xcrun xcodebuild -h` but not the man page.